src/economy/economy.cpp

```diff
@@ -11,6 +11,39 @@
 
 namespace economy {
 
+enum class economy_reason {
+	pop, factory, rgo, artisan, construction, nation, stockpile, overseas_penalty
+};
+
+std::string reason_to_string(economy_reason reason) {
```

Comment:
since these are all static constants, it may make more sense to return a `std::string_view`